src/client/src/ui/tearoff/BrowserPortal.tsx

```diff
@@ -0,0 +1,201 @@
+import * as React from 'react'
```

Comment:
These two implementations need to be cleaned up. Factor out what is common and put them in another file